src/vsr/replica.zig

```diff
@@ -2288,10 +2411,6 @@ pub fn ReplicaType(
                         });
                         self.write_prepare(next.message, .append);
                     }
-                } else {
-                    // When the pipeline is empty, stop the prepare timeout.
-                    // The timeout will be restarted when another entry arrives for the pipeline.
-                    self.prepare_timeout.stop();
```

Comment:
This timeout used to run when the pipeline was non-empty.
Now this timeout runs when the pipeline is non-empty _and_ at least one prepare in the pipeline is awaiting a quorum of acks (i.e. `!prepare.ok_quorum_received`).